### PR TITLE
Fix redirect limit in update check

### PR DESCRIPTION
**Pull request by AkibaAT**
## The Problem/Issue/Bug:
Resolves issue #326

## How this PR Solves the Problem:
redirectLimit has a misleading name, and actually requires a value of 1 + redirect limit.

See relevant HttpRequests code at https://github.com/JetBrains/intellij-community/blob/401226d2da0fff5e7996911fb2370cff7f66180a/platform/ide-core/src/com/intellij/util/io/HttpRequests.java#L552
and IntelliJ Community test suite at https://github.com/JetBrains/intellij-community/blob/451d49f1dde29a396096d1ed7c761c6afab62e68/platform/platform-tests/testSrc/com/intellij/util/io/HttpRequestsTest.kt#L89

## Manual Testing Instructions:
./gradlew build
Install resulting build/distributions/ddev-intellij-plugin-0.0.1-dev.zip
No more redirect exception on program start

### CODE REVIEW

**Comment by thomaskieslich** [comment]
I add the change from GithubClient.java to 2024.2 pull request and it works without Errors.
Tested on Sonoma 14.6.1.
If i should test more changes on Sonoma and phpstorm 2024.2 perhaps i can help.
Thanks for your work.

File: src/main/java/de/php_perfect/intellij/ddev/version/GithubClient.java [hunk 1]
```diff
@@ -19,7 +19,7 @@ public final class GithubClient implements ReleaseClient {
 
     @Override
     public @Nullable LatestRelease loadCurrentVersion(@NotNull ProgressIndicator indicator) {
-        final RequestBuilder requestBuilder = HttpRequests.request(RELEASE_URL).accept("application/json").redirectLimit(1);
+        final RequestBuilder requestBuilder = HttpRequests.request(RELEASE_URL).accept("application/json").redirectLimit(2);
```
Comment (nico-loeber) [inline src/main/java/de/php_perfect/intellij/ddev/version/GithubClient.java:22]: Did you find out why the limited needs to be increased?

**Review by nico-loeber — APPROVED** [review]

File: src/main/java/de/php_perfect/intellij/ddev/version/GithubClient.java (hunk 1 above)
Comment (AkibaAT) [reply src/main/java/de/php_perfect/intellij/ddev/version/GithubClient.java:22]: I did not look into the history in detail, but this commit from 8 months ago looks like the most probable cause: https://github.com/JetBrains/intellij-community/commit/b793719aac02659cef47aad371873bff0a15f51f

Comment (nico-loeber) [reply src/main/java/de/php_perfect/intellij/ddev/version/GithubClient.java:22]: Okay, looks like the additional redirect of github was ignored before this commit, yes.
Next thing is to find out why that redirect is happening, but not relevant for this pull request.
Will merge it, as soon as the tests are stable. :)

Comment (AkibaAT) [reply src/main/java/de/php_perfect/intellij/ddev/version/GithubClient.java:22]: There is only one redirect from latest to the current version, no additional redirects.
```
private static URLConnection openConnection(RequestBuilderImpl builder, RequestImpl request) throws IOException {
    if (builder.myForceHttps && request.myUrl.startsWith("http:")) {
      request.myUrl = "https:" + request.myUrl.substring(5);
    }

    for (int i = 0; i < builder.myRedirectLimit; i++) {
```

Unless I'm somehow reading this super wrong, the with a theoretical myRedirectLimit of 0 (which can't be set, the minimum is 1) you'd not get any connection.
You need at least 1 to get a connection to a page without redirects, and 2 if there is 1 redirect. Always 1 + expected redirects.

Comment (nico-loeber) [reply src/main/java/de/php_perfect/intellij/ddev/version/GithubClient.java:22]: Ahh, I love off by one errors! :D
Thanks!

**Pull request merged**